scripts/verify-tsconfig.mjs

```diff
@@ -0,0 +1,28 @@
+import { promises as fs } from 'fs';
+import pathUtils from 'path';
+import { fileURLToPath } from 'url';
+
+const cwd = pathUtils.dirname(fileURLToPath(import.meta.url))
+
+// These are the packages we expect to _not_ be referenced in the root tsconfig.
+const IGNORE_LIST = new Set(['examples', 'types']);
+
+// Get reference paths from root tsconfig.json
+const rootTsconfig = JSON.parse(await fs.readFile('./tsconfig.json', { encoding: 'utf8'}));
+const rootTsconfigReferences = new Set(rootTsconfig.references.map(
+  ({ path }) => path.split('/').pop()
+))
+
+// Get the names of all directories in the packages directory
+const packagesPath = pathUtils.resolve(cwd, '../packages');
```

Comment:
Might as well parse workspaces from the root package.json

Reply:
True, I was just lazy. I don't expect the workspace directory to ever change, and if it does, hopefully we'll remember this script.